src/test.rs

```diff
@@ -122,11 +122,20 @@ fn test_frame_allocator_add() {
 }
 
 #[test]
-#[should_panic]
-fn test_frame_allocator_add_large_size_panics() {
+fn test_frame_allocator_allocate_large() {
+    let mut frame = FrameAllocator::<32>::new();
+    assert_eq!(frame.alloc(10_000_000_000), None);
+}
+
+#[test]
+fn test_frame_allocator_add_large_size_split() {
     let mut frame = FrameAllocator::<32>::new();
 
     frame.insert(0..10_000_000_000);
+
+    assert_eq!(frame.alloc(0x8000_0001), None);
+    assert_eq!(frame.alloc(0x8000_0000), Some(0x8000_0000));
+    assert_eq!(frame.alloc(0x8000_0000), Some(0x10000_0000));
```

Comment:
Because the 10 GB range inserted into the allocator has been split into 2 GiB subranges (plus some smaller ones), and so there are enough to allocate several.

The fact that the first allocation starts at 0x8000_0000 rather than 0 seems to be because of the logic with `lowbit` in `FrameAllocator::add_frame`. I'm not really sure what's going on there, but I assume there is some reason for it so I didn't change it.